client/payment-details/payment-method/index.js

```diff
@@ -11,25 +11,33 @@ import { Card, CardBody, CardHeader } from '@wordpress/components';
  */
 import Loadable from 'components/loadable';
 import CardDetails from './card';
+import CardPresentDetails from './card-present';
 import GiropayDetails from './giropay';
 import SepaDetails from './sepa';
 import SofortDetails from './sofort';
 
 const PaymentDetailsPaymentMethod = ( { charge = {}, isLoading } ) => {
-	let PaymentMethodDetails = CardDetails;
+	const DetailComponents = {
+		card: CardDetails,
+		card_present: CardPresentDetails, // eslint-disable-line camelcase
+		giropay: GiropayDetails,
+		sepa_debit: SepaDetails, // eslint-disable-line camelcase
+		sofort: SofortDetails,
+	};
+
+	let PaymentMethodDetails = null;
 	if ( charge.payment_method_details && charge.payment_method_details.type ) {
-		switch ( charge.payment_method_details.type ) {
-			case 'giropay':
-				PaymentMethodDetails = GiropayDetails;
-				break;
-			case 'sepa_debit':
-				PaymentMethodDetails = SepaDetails;
-				break;
-			case 'sofort':
-				PaymentMethodDetails = SofortDetails;
-				break;
+		const type = charge.payment_method_details.type;
+		if ( type in DetailComponents ) {
+			PaymentMethodDetails = DetailComponents[ type ];
 		}
 	}
+
+	// Gracefully degrade for unrecognized payment method types
+	if ( null == PaymentMethodDetails ) {
+		return null;
+	}
```

Comment:
🔍 I think guard expression would make it even more readable:
```js
if ( ! charge.payment_method_details || ! charge.payment_method_details.type || ! type in detailsComponentMappings ) {
  return null;
}
const PaymentMethodDetails = detailsComponentMappings[charge.payment_method_deatils.type];
```

Reply:
Agreed. With some changes, added in https://github.com/Automattic/woocommerce-payments/pull/1840/commits/7b0ff25db73bcf423b01448f7deb6f74477a9740